src/apps/service/apps/de/jobs/resources.clj

```diff
@@ -0,0 +1,56 @@
+(ns apps.service.apps.de.jobs.resources
+  (:require [apps.util.config :as config]))
+
+(defn- get-min-resource-setting
+  "Determines the minimum amount of a specific resource required for the container."
+  [min-kw max-kw default container requirements]
+  (let [c-min (get container min-kw)
+        c-max (get container max-kw)
+        r-min (get requirements min-kw)
+        r-max (get requirements max-kw)]
+    (cond
+      ;; A nonexisent or nonsensical request defaults to the minimum defined for the container or the overall default.
+      (or (nil? r-min) (zero? r-min) (neg? r-min))
+      (or c-min default)
+
+      ;; If the requested minimum is less than the container minimum, default to the container's minimum setting.
+      (and c-min (< r-min c-min))
+      c-min
+
+      ;; In all other cases, the requested minimum must not be greater than the requested or container maximum.
+      :else
+      (apply min (filter pos? (remove nil? [r-min r-max (or c-max default)]))))))
+
+(defn- get-max-resource-setting
+  "Determines the maximum amount of a specific resource required for the container."
+  [max-kw default min-resource-setting-fn container requirements]
+  (let [c-max (get container max-kw)
+        r-max (get requirements max-kw)]
+    (cond
+      ;; If the requested maximum wasn't specified, use the requested minimum.
+      (or (nil? r-max) (zero? r-max) (neg? r-max))
+      (min-resource-setting-fn container requirements)
+
+      ;; If the requested maximum was specified, it must not be greater than the container maximum.
+      :else
+      (min (or c-max default) r-max))))
```

Comment:
This is where the bulk of the logic is for this project, so this is where I'd like the most scrutiny.